client/beefy/src/communication/request_response/incoming_requests_handler.rs

```diff
@@ -132,12 +138,36 @@ where
 		genesis_hash: Hash,
 		fork_id: Option<&str>,
 		client: Arc<Client>,
+		prometheus_registry: Option<prometheus::Registry>,
 	) -> (Self, RequestResponseConfig) {
 		let (request_receiver, config) =
 			on_demand_justifications_protocol_config(genesis_hash, fork_id);
 		let justif_protocol_name = config.name.clone();
+		let metrics = prometheus_registry
+			.as_ref()
+			.map(OnDemandIncomingRequestsMetrics::register)
+			.and_then(|result| match result {
+				Ok(metrics) => {
+					debug!(
+						target: "beefy",
+						"🥩 Registered on-demand incoming justification requests metrics"
+					);
+					Some(metrics)
+				},
+				Err(err) => {
+					debug!(
+						target: "beefy",
+						"🥩 Failed to register incoming justification requests metrics: {:?}",
+						err
+					);
+					None
+				},
+			});
```

Comment:
Just 2 nits:
1. On `Ok` we log `Registered on-demand incoming justification ...` and on `Err` we log `Failed to register incoming justification ...` (the `on-demand` is missing). The same for `outgoing_requests_engine.rs.rs`. Is this desired ?
2. This code seems to be duplicated in a couple of places. Maybe we could have it in a method. Or in a trait.

Reply:
done